src/lib.rs

```diff
@@ -245,11 +245,11 @@ pub trait Tree<H>: Frontier<H> {
     /// the specified leaf value from the specified position in the tree.
     fn is_witnessed(&self, position: Position, value: &H) -> bool;
 
-    /// Marks the current leaf as one for which we're interested in producing
-    /// an authentication path. Returns an optional value containing the current
-    /// position if successful or if the current value was already marked,
-    /// or None if the tree is empty.
-    fn witness(&mut self) -> bool;
+    /// Marks the current leaf as one for which we're interested in producing 
```

Comment:
```suggestion
    /// Marks the current leaf as one for which we're interested in producing
```
I also don't understand why the documentation change for this function is present: it seems like the documentation was incorrect before but correct now?